documents/state_machine.py

```diff
@@ -0,0 +1,62 @@
+from config import config
+from models.db import db
+from models.User import User
+import jwt
+from models.Document import Document
+
+def get_user_by_user_id(user_id):
+    try:
+        users = db.session.query(User).filter(User.id == user_id)
+        if users.count() <= 0:
+            return None
+        if users.count() == 1:
+            for i in users:
+                return vars(i)
+        if users.count() > 1:
+            raise Exception({'error':'voilates the unique ability!'})
+    except Exception as e:
+        raise Exception({'error':str(e.message)})
+    
+def upsert_document(user_id):
+    try:
+        document = Document(
+            user_id = user_id
+        )
+        db.session.add(document)
+        db.session.commit()
+        return document.id
+    except Exception as e:
+        raise Exception({'error':str(e.message)})
+
+def get_document_by_id(id):
+    try:
+        documents = db.session.query(Document).filter(Document.id == id)
+        if documents.count() <= 0:
+            return None
+        if documents.count() == 1:
+            for i in documents:
+                return vars(i)
+        if documents.count() > 1:
+            raise Exception({'error':'voilates the unique ability!'})
+    except Exception as e:
+        raise Exception({'error':str(e.message)})
+
+def token_valid_check(auth_header):
```

Comment:
why is this token_valid_check re-written in this documents/state_machine file? 
there should be only one token validity check function for the entire app, not new function for every module you write

Reply:
This one uses an authorization header, the other one takes in token as a param

Reply:
I don't understand what that means

Reply:
then this is common for all APIs no, not just documents. It should not be in this state_machine